### PR TITLE
Add codecov.yml

### PR DESCRIPTION
**Pull request by pitrou**
Codecov.io migrated to a new version where the settings must be stored in a YAML file
(rather than entered in the Web UI).

### CODE REVIEW

**Comment by pitrou** [comment]
Uh, apparently codecov decided not to post any issue comment...

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Configuration for codecov.io
+
+comment:
+    layout: "header, changes, diff"
+
+coverage:
+    ignore:
+        - numba/cuda/.*
```
Comment (stuartarchibald) [inline codecov.yml]: it seems to not be ignoring this? https://codecov.io/gh/numba/numba/list/96c18dd50a4f155f9fe23df57f1f026c0e8e696d then sort by coverage

Comment (pitrou) [inline codecov.yml]: Yes, that's the problem I'm trying to solve.

**Comment by pitrou** [comment]
I recommend keeping this PR open until the current `codecov.io` issues resolve, and until we find out how to make those configuration directives work.

**Comment by stevepeak** [comment]
Hey @pitrou deploying a fix for this in the next hour. Thank you for your patience. 👍

**Comment by pitrou** [comment]
Ok, it seems it's fixed now.

**Pull request merged**